plugins/mcreator-core/blockly/mcreator_extensions.js

```diff
@@ -125,6 +125,18 @@ Blockly.Extensions.register('throwableprojectile_list_provider',
                 javabridge.getReadableListOf("throwableprojectile"))), 'throwableprojectile');
     });
 
+Blockly.Extensions.register('custom_entity_with_entity_data_list_provider',
+    function () {
+        this.appendDummyInput()
+            .appendField(new Blockly.FieldDropdown(
+            arrayToBlocklyDropDownArrayWithReadableNames(javabridge.getListOf("customentity"),
+                javabridge.getReadableListOf("entity"))), 'customentity');
+        this.appendDummyInput()
+                .appendField(new Blockly.FieldDropdown(
+                    arrayToBlocklyDropDownArrayWithReadableNames(javabridge.getCustomEntityDataList(this.getFieldValue('customentity')),
+                        javabridge.getReadableListOfWithValue('entitydata', this.getFieldValue('customentity')))), 'dataName');

```

Comment:
I've checked a bunch of Blockly source files and I think that the dropdown may be properly updated without any callback functions. Can you please try redefining `dataName` field like this and tell if that resolves the problem? (I have nothing to test this on, sorry)
```suggestion
        this.appendDummyInput().appendField(new Blockly.FieldDropdown(
                function (fieldDropdown) {
                        return arrayToBlocklyDropDownArrayWithReadableNames(javabridge.getCustomEntityDataList(this.getFieldValue('customentity')),
                                javabridge.getReadableListOfWithValue('entitydata', this.getFieldValue('customentity')));
                }), 'dataName');
```

Reply:
This may help: https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/dropdown#creating_a_dropdown_validator

Reply:
Changed a single line of the code I suggested, @Goldorion could you please check if this updates the selected `'dataName'` value?